test: a11y refactoring

ravenokko opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us process your pull request.

Please make sure to fulfil our contribution guidelines (https://developer.shopware.com/docs/resources/guidelines/code/contribution?category=shopware-platform-dev-en/contribution).

Do your changes need to be documented?
Please create a second pull request at https://github.com/shopware/docs
-->

### 1. Why is this change necessary?

Updates tests to include keyboard functionality and a11y checks

### 2. What does this change do, exactly?

Refactoring of tests based on https://github.com/shopware/acceptance-test-suite/pull/410
All relevant tasks and locators have already been refactored with https://github.com/shopware/acceptance-test-suite/pull/491 which was included in [ATS v11.36.0](https://github.com/shopware/acceptance-test-suite/pull/506) and officially integrated into platform with [ATS v11.37.0](https://github.com/shopware/shopware/pull/13144)

### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).
<!-- Examples:
- closes #123  - closes the issue #123 when the PR is merged
- relates #123 - relates to the issue #123

If the issue exists only in Jira, include a link to the Jira issue.
- Jira issue: https://shopware.atlassian.net/browse/NEXT-123
-->

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfilled them

ravenokko committed:
test: update to include keyboard functionality and a11y checks

ravenokko committed:
test: update StorefrontCurrencies to include ChangeStorefrontCurrency functionality

ravenokko committed:
test: a11y compatible RegisteredUserBuysProduct

ravenokko committed:
test: update RegisteredUserBuysProduct

ravenokko committed:
Merge branch 'trunk' into test/a11y-cherry-pick

ravenokko committed:
merge trunk into cherry pick

ravenokko committed:
style: GoogleReCaptchaV2

ravenokko committed:
style: indentation cleaunup

ravenokko committed:
test: a11y WishlistGuestFunctionalities

ravenokko committed:
test: replace SelectInvoicePaymentOption with SelectPaymenthMethod('Invoice') and SelectStandardShippingOption with SelectShippingMethod('Standard')

ravenokko committed:
test: fix duplicate SelectPaymentMethods

ravenokko committed:
test: remove duplicate GoogleAnalyticsDeclineTracking

ravenokko committed:
test: fixed payment method from Paid in Advance to Invoice

ravenokko committed:
Change payment method to Invoice

This is the default configuration

ravenokko committed:
test: ChangeOrderPaymentMethod

ravenokko committed:
test: finalize ChangeOrderPaymentMethod

ravenokko committed:
test: GoogleReCaptchaV2 fix

ravenokko committed:
Merge branch 'test/a11y-cherry-pick' into test/a11y-integration

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
refactor: click() into presses()

ravenokko committed:
refactor: remove unnecessary key override from presses()

ravenokko committed:
refactor: more removing unnecessary keys overrides from presses()

ravenokko committed:
style: formatiing typos

ravenokko committed:
test: delete generated snapshots

ravenokko committed:
refactor: update DownloadDigitalProductFromOrder task

ravenokko committed:
test: update StorefrontCurrencies, BasicConsentBannerDeclineCookies and ValidateContactFormStorefront

ravenokko committed:
test: add check for visible focus in second ProductFilters test

moshimorschi commented on tests/acceptance/tests/Account/ForgotPassword.spec.ts:
```diff
@@ -65,17 +65,17 @@ test ('As a customer, I can reset my password using the password recovery proces
     });
 
     await test.step('Fill in the customer email and request a password reset', async () => {
         await StorefrontAccountRecover.emailInput.fill(customer.email);
```
```suggestion
        await ShopCustomer.fillsIn(StorefrontAccountRecover.emailInput, customer.email);
```

moshimorschi replied:
I just spotted this, perhaps it was intentional but just in case it was not, would this not be the current approach? :)

moshimorschi commented on tests/acceptance/tests/Account/ForgotPassword.spec.ts:
```diff
@@ -9,17 +9,17 @@ test ('As a customer, I can request a new password with existing customer email
     });
 
     await test.step('Fill in the customer email and request a password reset', async () => {
         await StorefrontAccountRecover.emailInput.fill(customer.email);
```
```suggestion
        await ShopCustomer.fillsIn(StorefrontAccountRecover.emailInput, customer.email);
```
marking the lines where the .fillsIn() method is missing :)

moshimorschi commented on tests/acceptance/tests/Account/ForgotPassword.spec.ts:
```diff
@@ -31,17 +31,17 @@ test ('As a customer, I can request a new password without existing customer ema
     });
 
     await test.step('Request password reset with a non-existing email', async () => {
         await StorefrontAccountRecover.emailInput.fill('test-forgot-password-non-existing@email.net');
```
```suggestion
        await ShopCustomer.fillsIn(StorefrontAccountRecover.emailInput, 'test-forgot-password-non-existing@email.net');
```
another one, replaced with .fillsIn() method as well!

ravenokko committed:
test: remove login wait which was transferred to LoginViaReviewsTab task

moshimorschi committed:
replacing .click()  with .presses() method

replaced two more .click() methods for submitting  contact form, that could be replaced with .presses() method.

moshimorschi commented on tests/acceptance/tasks/ShopCustomer/Account/DownloadDigitalProductFromOrder.ts:
```diff
@@ -2,15 +2,15 @@ import { test as base, expect } from '@playwright/test';
         const task = (contentOfFile: string) => {
             return async function DownloadDigitalProductFromOrder() {
                 // TODO: Migrate to StorefrontAccountOrder.orderExpandButton.click(); when https://github.com/shopware/acceptance-test-suite/pull/126 is released.
                 await StorefrontAccountOrder.page.locator('.order-hide-btn').first().click();
```
```suggestion
               await ShopCustomer.presses(StorefrontAccountOrder.page.locator('.order-hide-btn').first());
```

ravenokko committed:
test: ProductFilters fixed and locators updated for AcceptTechnicalRequiredCookies

ravenokko committed:
Merge branch 'test/a11y-integration' of github.com:shopware/shopware into test/a11y-integration

ravenokko committed:
Missing presses() and fillsIn()

Co-authored-by: Lena <118278183+moshimorschi@users.noreply.github.com>

ravenokko committed:
test: remove 'test.only' from ProfuctFilter and unnecessary visibility checks from ChangeCustomerCredentials

moshimorschi commented on tests/acceptance/tests/Account/GeoBlockingNonShippableCountry.spec.ts:
spotted one last elusive .click() method here:
` await StorefrontAccountAddresses.addressDropdownButtons.last().click();`

ravenokko committed:
test: remove more clicks, add comments and skips to ProductFilter and ProductFilter

ravenokko committed:
test: update ProductFilter with additional assertions for stability

ravenokko committed:
test: refactor VariantGeneration to use propertyRadioGroup

ravenokko committed:
test: update StorefrontCurrencies to USD default and Header locators

ravenokko committed:
test: update Wishlist tests to use Header locators

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
fix: missing ChangeStorefrontCurrency class

ravenokko committed:
test: update StorefrontLanguages for language agnostic locators

ravenokko committed:
fix: remove test.only()

ravenokko committed:
test: fix failing StorefrontCurrency test

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
refactor: click to presses in StorefrontLanguages

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
fix: updated StorefrontLanguages and StorefrontCurrencies to use StorefrontHeader locators

ravenokko committed:
fix: revert filter by rating test to use click due to technical limitations

ravenokko committed:
fix: add networkIdle wait for captcha to fully load

ravenokko committed:
fix: Language locator and toPass() for Captcha final test step

ravenokko committed:
style: remove unnecessary comment

ravenokko committed:
fix: use ProceedFromCartToCheckout in GuestCustomerRegistration

ravenokko committed:
style: add space to DownloadDigitalProductFromOrder task

ravenokko committed:
fix: flaky GuestCustomerRegistration.spec.ts where Playwright needed to wait at least until the register page started to load after pressing the Go To Checkout button

ravenokko committed:
fix: add scrollIntoView and fixed wrong GoToCheckout locator

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration

ravenokko committed:
Merge branch 'trunk' into test/a11y-integration
